doc: Add explicit macdeployqtplus dependencies install step

hebasto opened the pull request:
This PR adds to macOS docs an explicit step to install `macdeployqtplus` script dependencies that are not part of the [Python Standard Library](https://docs.python.org/3/library/index.html):
- https://pypi.org/project/ds-store/
- https://pypi.org/project/mac-alias/

This change is required on macOS 11 Big Sur:

-  #20371
- #20878

Close #20878.

hebasto committed:
doc: Add explicit macdeployqtplus dependencies install step

This change is required on macOS 11 Big Sur.

jarolrod commented:
ACK aaa7b294f8323b0087b090180819aeadcf42a826

Confirming that without installing `argparse`, `ds_store`, `mac_alias`, make deploy fails on macOS Big Sur 11.1.

Deploying DMG is successful after installing these packages with `pip3`.

jarolrod reviewed (commented):
ACK aaa7b294f8323b0087b090180819aeadcf42a826

Confirming that without installing `argparse`, `ds_store`, `mac_alias`, make deploy fails on macOS Big Sur 11.1.

Deploying DMG is successful after installing these packages with `pip3`.

hebasto commented on doc/build-osx.md:
````diff
@@ -30,6 +30,11 @@ If you want to build the disk image with `make deploy` (.dmg / optional), you ne
 
+and [`macdeployqtplus`](../contrib/macdeploy/README.md) dependencies:
+```shell
+pip3 install argparse ds_store mac_alias
````
> argparse is part of the Python standard library.

Indeed: https://docs.python.org/3/library/argparse.html

> It doesn't need installing.

Fixed.

hebasto commented:
Updated aaa7b294f8323b0087b090180819aeadcf42a826 -> 3e61b8c800180d350621cedda7ec46a48047ff04 ([pr20890.02](https://github.com/hebasto/bitcoin/commits/pr20890.02) -> [pr20890.03](https://github.com/hebasto/bitcoin/commits/pr20890.03), [diff](https://github.com/hebasto/bitcoin/compare/pr20890.02..pr20890.03)):

- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/20890#discussion_r554273503):
> argparse is part of the Python standard library. It doesn't need installing.

fanquake reviewed (approved):
ACK 3e61b8c800180d350621cedda7ec46a48047ff04

> This change is required on macOS 11 Big Sur:

There's nothing macOS 11 specific about this doc addition. The  same dependencies & dmg creation process is used you run `make deploy` on macOS 10.
